### PR TITLE
Promote ImageMaximumGCAge to beta

### PR DESCRIPTION
**Pull request by haircommander**
#### What type of PR is this?

/kind feature
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Promote ImageMaximumGCAge feature to beta
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/issue/4210
```

### CODE REVIEW

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -1153,7 +1154,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 
 	LoadBalancerIPMode: {Default: false, PreRelease: featuregate.Alpha},
 
-	ImageMaximumGCAge: {Default: false, PreRelease: featuregate.Alpha},
+	ImageMaximumGCAge: {Default: false, PreRelease: featuregate.Beta},
```
[the remark this answers is not in the document]
Comment (haircommander) [reply pkg/features/kube_features.go:1157]: I figured beta flags are off by default now so we'd leave it off for a release maybe but I'm open. WDYT @dchen1107 @SergeyKanzhelev @mrunalp

[the remark this answers is not in the document]
Comment (kannon92) [reply pkg/features/kube_features.go:1157]: TBH sig-node differs here from other features I worked on. Usually beta features are on by default (so people can actually use the feature). Alpha is off by default.

I think before we GA we would have to set this to true by default so its up to you on how much soak time you want for the feature. 

I think, unless this is a really risky feature, we should roll this out as true for beta.

[the remark this answers is not in the document]
Comment (haircommander) [reply pkg/features/kube_features.go]: sure works for me. It is technically off by default anyway, because you also need to configure a kubelet flag

**Comment by kannon92** [comment]
PR looks good. You addressed the feedback in the review in separate PRs so I think this is good to me.

/lgtm
/assign @mrunalp @dchen1107 @SergeyKanzhelev

**Comment by kannon92** [comment]
/triage accepted
since its part of a KEP
/priority important-soon
next week should be merged

**Review by mrunalp — APPROVED** [review]

**Comment by kannon92** [comment]
/label api-review
/assign @liggitt 

Very minor issue in kubelet config tests and we need api approver help.

File: pkg/features/kube_features.go [hunk 2]
```diff
@@ -1153,7 +1154,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 
 	LoadBalancerIPMode: {Default: false, PreRelease: featuregate.Alpha},
 
-	ImageMaximumGCAge: {Default: false, PreRelease: featuregate.Alpha},
+	ImageMaximumGCAge: {Default: true, PreRelease: featuregate.Beta},
```
Comment (liggitt) [inline pkg/features/kube_features.go:1157]: move this line so it is in alphabetical order in the list of features, that helps minimize conflicts

File: pkg/kubelet/apis/config/validation/validation_test.go [hunk 3]
```diff
@@ -526,22 +526,21 @@ func TestValidateKubeletConfiguration(t *testing.T) {
 	}, {
 		name: "imageMaximumGCAge should not be specified without feature gate",
 		configure: func(conf *kubeletconfig.KubeletConfiguration) *kubeletconfig.KubeletConfiguration {
+			conf.FeatureGates = map[string]bool{"ImageMaximumGCAge": false}
```
Comment (liggitt) [inline pkg/kubelet/apis/config/validation/validation_test.go:529]: adding this line is good / necessary

File: pkg/kubelet/apis/config/validation/validation_test.go [hunk 4]
```diff
@@ -526,22 +526,21 @@ func TestValidateKubeletConfiguration(t *testing.T) {
...
 		configure: func(conf *kubeletconfig.KubeletConfiguration) *kubeletconfig.KubeletConfiguration {
-			conf.FeatureGates = map[string]bool{"ImageMaximumGCAge": true}
 			conf.ImageMaximumGCAge = metav1.Duration{Duration: -1}
 			return conf
 		},
 		errMsg: "invalid configuration: imageMaximumGCAge -1ns must not be negative",
 	}, {
 		name: "imageMaximumGCAge should not be less than imageMinimumGCAge",
 		configure: func(conf *kubeletconfig.KubeletConfiguration) *kubeletconfig.KubeletConfiguration {
-			conf.FeatureGates = map[string]bool{"ImageMaximumGCAge": true}
```
Comment (liggitt) [inline pkg/kubelet/apis/config/validation/validation_test.go:544]: keep these lines... if we have to toggle the gate back off, we don't want to have to mess with this unit test to test this behavior

File: test/e2e_node/image_gc_test.go [hunk 5]
```diff
@@ -57,10 +56,6 @@ var _ = SIGDescribe("ImageGarbageCollect", framework.WithSerial(), framework.Wit
 		tempSetCurrentKubeletConfig(f, func(ctx context.Context, initialConfig *kubeletconfig.KubeletConfiguration) {
 			initialConfig.ImageMaximumGCAge = metav1.Duration{Duration: time.Duration(time.Minute * 1)}
 			initialConfig.ImageMinimumGCAge = metav1.Duration{Duration: time.Duration(time.Second * 1)}
-			if initialConfig.FeatureGates == nil {
-				initialConfig.FeatureGates = make(map[string]bool)
-			}
-			initialConfig.FeatureGates[string(kubefeatures.ImageMaximumGCAge)] = true
```
Comment (liggitt) [inline test/e2e_node/image_gc_test.go:63]: does it hurt anything to keep the feature explicitly turned on in the test?

**Comment by haircommander** [comment]
thanks @liggitt done!

**Pull request merged**